static/js/panels.js

```diff
@@ -101,24 +104,101 @@ export function show_profile_incomplete(is_profile_incomplete) {
     } else if (page_params.server_needs_upgrade) {
         if (should_show_server_upgrade_notification(ls)) {
-            open($("[data-process='server-needs-upgrade']"));
+            open({
+                data_process: $t({defaultMessage: "server-needs-upgrade"}),
+                class_name: $t({defaultMessage: "red"}),
+                first_step_message: $t({
+                    defaultMessage:
+                        "This Zulip server is running an old version and should be upgraded.",
+                }),
+                first_step_buttons: [
+                    {
+                        text: $t({defaultMessage: "Learn more"}),
+                        link: "https://zulip.readthedocs.io/en/latest/overview/release-lifecycle.html#upgrade-nag",
+                    },
+                    {
+                        class_name: $t({defaultMessage: "dismiss-upgrade-nag"}),
+                        text: $t({defaultMessage: "Dismiss for a week"}),
+                    },
+                ],
+            });
         }
     } else if (page_params.warn_no_email === true && page_params.is_admin) {
         // if email has not been set up and the user is the admin,
         // display a warning to tell them to set up an email server.
-        open($("[data-process='email-server']"));
+        open({
+            data_process: $t({defaultMessage: "email-server"}),
+            class_name: $t({defaultMessage: "red"}),
+            first_step_message: $t({
+                defaultMessage:
+                    "Zulip needs to send email to confirm users' addresses and send notifications.",
+            }),
+            first_step_button_text: $t({defaultMessage: "See how to configure email."}),
+            first_step_button_link: "https://zulip.readthedocs.io/en/latest/production/email.html",
+        });
     } else if (should_show_notifications(ls)) {
-        open($("[data-process='notifications']"));
+        show_notification_alert();
     } else if (unread_ui.should_display_bankruptcy_banner()) {
-        open($("[data-process='bankruptcy']"));
+        open({
+            data_process: $t({defaultMessage: "bankruptcy"}),
+            class_name: $t({defaultMessage: "bankruptcy"}),
+            first_step_message: $t({defaultMessage: "Welcome back! You have"}),
+            first_step_remaing_message: $t({
+                defaultMessage: " unread messages. Do you want to mark them all as read?",
+            }),
+
+            is_bankruptcy_alert: true,
+            unread_msgs_count: page_params.unread_msgs.count,
+            first_step_buttons: [
+                {
+                    class_name: $t({defaultMessage: "accept-bankruptcy"}),
+                    text: $t({defaultMessage: "Yes, please!"}),
+                },
+                {
+                    class_name: $t({defaultMessage: "exit"}),
+                    text: $t({defaultMessage: "No, I'll catch up"}),
+                },
+            ],
+        });
```

Comment:
This is a ton of configuration and feels very hard to read/review; can you think about other ways to do the interface between this and the template?